code/modules/urist/datums/xenoarch_effects/extinguisher.dm

```diff
@@ -0,0 +1,47 @@
+/datum/artifact_effect/extinguisher
+	name = "Extinguisher"
+	effect_type = EFFECT_UNKNOWN
+	effect_icon = 'icons/effects/effects.dmi'
+	effect_state = "splash"
+
+
+/datum/artifact_effect/extinguisher/New()
+	. = ..()
+	effect = pick(EFFECT_PULSE, EFFECT_TOUCH)
+
+
+/datum/artifact_effect/extinguisher/DoEffectTouch(mob/living/user)
+	if(user)
+		if(istype(user, /mob/living/carbon/human))
+			user.ExtinguishMob()
+			user.fire_stacks = clamp(user.fire_stacks - 1, -25, 25)
```

Comment:
Should just do a water_act() on the user atom too if we're acting like this actually produces moisture. It will already handle fire stuff as well.
```suggestion
	if(user)
	        user.water_act(2)
```